src/components/Table/TableRow.tsx

```diff
@@ -1,25 +1,24 @@
-import { useProjectContext } from "@src/contexts/ProjectContext";
+import { useSetAnchorEle } from "@src/atoms/ContextMenu";
 import { Fragment } from "react";
 import { Row, RowRendererProps } from "react-data-grid";
 
 import OutOfOrderIndicator from "./OutOfOrderIndicator";
 
 export default function TableRow(props: RowRendererProps<any>) {
-  const { contextMenuRef }: any = useProjectContext();
+  const { setAnchorEle } = useSetAnchorEle();
   const handleContextMenu = (
     e: React.MouseEvent<HTMLDivElement, MouseEvent>
   ) => {
     e.preventDefault();
-    if (contextMenuRef?.current) contextMenuRef?.current?.setAnchorEl(e.target);
+    setAnchorEle?.(e?.target as HTMLElement);
```

Comment:
```suggestion
    setAnchorEle(e?.target as HTMLElement);
```

I thought `setAnchorEle` would always exist/never be `undefined`

Reply:
You are right. Ill change it